plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/animators/PuckAnimator.kt

```diff
@@ -25,7 +26,9 @@ internal abstract class PuckAnimator<T>(
     setEvaluator(evaluator)
     addUpdateListener {
       @Suppress("UNCHECKED_CAST")
-      updateLayer(it.animatedFraction, it.animatedValue as T)
+      val updatedValue = it.animatedValue as T
+      updateLayer(it.animatedFraction, updatedValue)
+      puckUpdateListener?.invoke(updatedValue)
```

Comment:
Should `puckUpdateListener` be reset when an animation is complete?

Reply:
No, it actually should not matter.
All this API is internal and actually only thing we need is proxy updated value up to `LocationPuckManager` to always have latest value there.

Reply:
@Chaoba reworked approach, PTAL